Mail/Views/Search/SearchFilterFolderCell.swift

```diff
@@ -50,50 +56,53 @@ struct SearchFilterFolderCell: View {
         icon: MailResourcesAsset.allFolders.swiftUIImage
     )
 
-    private var sortedFolders: [Folder] {
-        return folders.filter(\.shouldBeDisplayed).sorted()
-    }
-
     var body: some View {
-        Menu {
-            Button {
-                withAnimation {
-                    selectedFolderId = allFoldersItem.id
-                }
-            } label: {
-                HStack {
-                    allFoldersItem.icon
-                    Text(allFoldersItem.name)
-                }
+        HStack(spacing: IKPadding.searchFolderCellSpacing) {
+            if isSelected {
+                MailResourcesAsset.check
+                    .iconSize(.small)
             }
-
-            ForEach(sortedFolders) { folder in
-                Button {
-                    withAnimation {
-                        selectedFolderId = folder.remoteId
-                    }
-                } label: {
-                    HStack {
-                        folder.icon
-                        Text(folder.localizedName)
-                    }
-                }
+            Text(selectedFolderName)
+                .font(MailTextStyle.bodyMedium.font)
+            ChevronIcon(direction: .down, shapeStyle: HierarchicalShapeStyle.primary)
+        }
+        .filterCellStyle(isSelected: isSelected)
+        .onTapGesture {
+            isShowingFolderList.toggle()
+        }
+        .sheet(isPresented: $isShowingFolderList) {
+            SearchableFolderListView(
```

Comment:
To keep a lightweight view and clean body, can we move the content of the sheet in a separate struct and file?